packages/headless/src/features/commerce/pagination/pagination-slice.ts

```diff
@@ -91,20 +99,24 @@ export const paginationReducer = createReducer(
       .addCase(registerRecommendationsSlotPagination, (state, action) => {
         const slotId = action.payload.slotId;
 
-        if (slotId in state) {
+        if (slotId in state.recommendations) {
           return;
         }
 
         state.recommendations[slotId] = getCommercePaginationInitialSlice();
       })
       .addCase(restoreSearchParameters, handleRestoreParameters)
       .addCase(restoreProductListingParameters, handleRestoreParameters)
-      .addCase(deselectAllBreadcrumbs, handlePaginationReset)
-      .addCase(deselectAllFacetValues, handlePaginationReset)
+      .addCase(clearAllCoreFacets, handlePaginationReset)
+      .addCase(deselectAllValuesInCoreFacet, handlePaginationReset)
       .addCase(toggleSelectFacetValue, handlePaginationReset)
       .addCase(toggleExcludeFacetValue, handlePaginationReset)
       .addCase(toggleSelectNumericFacetValue, handlePaginationReset)
       .addCase(toggleExcludeNumericFacetValue, handlePaginationReset)
+      .addCase(toggleSelectDateFacetValue, handlePaginationReset)
+      .addCase(toggleExcludeDateFacetValue, handlePaginationReset)
+      .addCase(toggleSelectCategoryFacetValue, handlePaginationReset)
+      .addCase(applySort, handlePaginationReset)
```

Comment:
Those were missing cases.

The toggle actions were not handled at all.

In the case of `applySort`, a pagination action to set page to 0 was dispatched through the sort controller (see https://github.com/coveo/ui-kit/pull/4124/files#diff-ae2ae8b42c29c5393a652260ee3de614c44f4cddb62c26d63134c487e6900512L145). I think it makes more sense that applySort would be handled in the pagination reducer.